api/src/main/java/marquez/db/migrations/V44_1__BackfillAirflowParentRuns.java

```diff
@@ -0,0 +1,193 @@
+package marquez.db.migrations;
+
+import java.util.List;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import marquez.common.Utils;
+import org.flywaydb.core.api.MigrationVersion;
+import org.flywaydb.core.api.migration.Context;
+import org.flywaydb.core.api.migration.JavaMigration;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.result.ResultProducers;
+
+/**
+ * This migration is dependent on the migration script in V44__runs_job_versions_add_job_uuid.sql.
+ * The version returned by this code is intentionally after V44, but before V45 so that casual
+ * contributors won't accidentally create a migration script that conflicts with the version of this
+ * one.
+ *
+ * <p>This migration backfills parent jobs and runs for Airflow DAGs that reported ParentRunFacets
+ * that used the task name as the job name and the scheduled run name as the runId. For example,
+ * many Airflow tasks report the ParentRunFacet as <code>
+ *   "parentRun": {
+ *         "job": {
+ *           "name": "simple_shortrunning_dag.tasks__1_of_3",
+ *           "namespace": "abcdefg"
+ *         },
+ *         "run": {
+ *           "runId": "scheduled__2022-03-14T01:40:10+00:00"
+ *         }
+ *       }
+ * </code> The RunId can't be mapped to an existing Marquez run and the job name is actually a
+ * concatenation of the DAG name and the task (which is the name assigned to the job for the task
+ * itself).
+ *
+ * <p>This migration aims to create a real parent run, with a distinct run id, and parent job for
+ * each such run by constructing a deterministic run id from the DAG name and its runId. From each
+ * run, a job is created with the name field set to the DAG name.
+ */
+@Slf4j
+public class V44_1__BackfillAirflowParentRuns implements JavaMigration {
+
+  /**
+   * Return a numeric version that is greater than 44 (so it executes after that one) but before 45
+   */
+  public static final MigrationVersion MIGRATION_VERSION = MigrationVersion.fromVersion("44.1");
+
+  private static final String FIND_AIRFLOW_PARENT_RUNS_SQL =
+      """
+      SELECT DISTINCT(run_uuid) AS run_uuid,
+                     e.parent_run_id,
```

Comment:
Assuming you branched off PR [#1935](https://github.com/MarquezProject/marquez/pull/1935) to continue your work before the PR was merge, but just wanted to point out you'll want to use `parent_run_uuid ` and `parent_job_uuid` 😅  (see my earlier [comment](https://github.com/MarquezProject/marquez/pull/1935#discussion_r845583636))

Reply:
Oh wait, this is the OpenLineage event!

Reply:
Oh wait, this is the OpenLineage event!